mirrord/layer/src/file/hooks.rs

```diff
@@ -336,6 +336,31 @@ pub(crate) unsafe extern "C" fn _read_nocancel_detour(
         .unwrap_or_bypass_with(|_| FN__READ_NOCANCEL(fd, out_buffer, count))
 }
 
+#[hook_guard_fn]
+pub(crate) unsafe extern "C" fn read_nocancel_detour(
```

Comment:
Is this the exact same code as `_read_nocancel_detour`? Do we actually still need `_read$NOCANCEL`? Does it really exist? If we do - can't we use the same detour function for both hooks?

Reply:
Not sure about the first question and we can't share because one might not resolve so the other would fail

Reply:
> one might not resolve so the other would fail

What does this mean? How does a detour function _resolve_?
My suggestion is to replace both symbols (with and without a preceding `_`) with the same detour function. Meaning jump from both libc functions to the same mirrord function. Can you explain what would fail there?

Reply:
In casw of a bypass, it would try to call the original. If the original is the other function and it doesn't exist it would fail

Reply:
Ahhh, I see, makes sense. Thanks!